crates/prices/src/lib.rs

```diff
@@ -1,17 +1,33 @@
-use {api::Api, std::time::Duration};
+use {api::Api, clap::Parser, std::time::Duration};
 
 mod api;
 mod boundary;
+mod cli;
+mod config;
 mod core;
 mod util;
 
-pub async fn main() {
+pub async fn main(args: impl Iterator<Item = String>) {
+    let args = cli::Args::parse_from(args);
+    let config = config::load(&args.config).await;
     let (shutdown_sender, shutdown_receiver) = tokio::sync::oneshot::channel();
+
+    let mut estimators: Vec<Box<dyn core::Estimator>> = Vec::new();
+    if let Some(zeroex) = config.zeroex {
+        if zeroex.enable {
+            estimators.push(Box::new(
+                boundary::Zeroex {
+                    api_key: zeroex.api_key,
+                    endpoint: zeroex.endpoint,
+                    timeout: std::time::Duration::from_millis(config.timeout_ms),
```

Comment:
What's the reason to move `timeouts` into each individual estimator instead of having it in a single component that queries all of them?
Seems a bit error prone.